### PR TITLE
Milestone 2: Adding data-media-background prop

### PR DESCRIPTION
**Pull request by tingting-lwy**
## Milestone 2: Adding the page-color-scheme

- Added the property data-media-background that takes the JSON object MediaBackgroundConfig
- Added MediaBackgroundConfig and its values to MarketplaceView
- Created a group of fields on the Configurator for the data-media-background attribute
- Users can test an image, colors (or multiple), and gradient types and angles

### Edge Cases tested:
- an Image URI should take precedence over colors
- if image URI is broken/wrong, fallback color takes over
- if no gradient type is selected and there is multiple colors, no background is used (assumed expected behavior)
- Changing angle or gradient type has no effect on single color or image backgrounds

Here is an example of an image
image

Here is an example of a singular color:
image

Here is an example of gradient colors:
image

Here is an example of gradient colors using other CSS valid values:
image

### CODE REVIEW

**Review by jaxonL — COMMENTED** [review]
works as intended, with a minor bug on the generated code.

File: vue/configurator/src/components/Configurator.vue [hunk 1]
```diff
@@ -285,6 +374,11 @@ export default class Configurator extends Vue {
             .replaceAll(/data-v-[a-z0-9]*="" ?/g, "")
             .replace(" >", ">")
             .replaceAll("  ", " ") ?? "";
+        // Replace all instances of HTML entity &quot; with `"`
+        // When a JSON object is JSON.stringify(), HTML text will encode double quotes to avoid misinterpretation
+        // Converting this back to double quotes solves user issue's with directly copy pasting the div element
+        // https://en.wikipedia.org/wiki/List_of_XML_and_HTML_character_entity_references
+        outputDiv.innerText = outputDiv.innerText.replaceAll("&quot;", '"');
```
[the remark this answers is not in the document]
Comment (tingting-lwy) [reply vue/configurator/src/components/Configurator.vue:381]: Oh gotcha! So if my understanding is correct, &quot; being displayed in the Body section on the configurator page is necessary?

File: vue/configurator/src/components/lib/WidgetProps.ts [hunk 2]
```diff
@@ -12,7 +29,7 @@ interface WidgetPropOptions {
 export interface WidgetPropDefinition {
   name: string;
   type: WidgetPropType;
-  value: string | boolean;
+  value: string | boolean | MediaBackgroundConfig;
   defaultValue: string | boolean;
```
[the remark this answers is not in the document]
Comment (tingting-lwy) [reply vue/configurator/src/components/lib/WidgetProps.ts]: good catch! I missed that, updated now!

File: vue/configurator/src/components/Configurator.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (tingting-lwy) [reply vue/configurator/src/components/Configurator.vue]: Updated to remove the string replace

File: vue/configurator/src/components/Configurator.vue [hunk 3]
```diff
@@ -68,6 +68,72 @@
...
+                  class="w-50 m-2"
+                  :placeholder="key"
+                />
+                <div
+                  v-else-if="
+                    typeof config === 'object' &&
+                    config?.type === WidgetPropType.ARRAY
+                  "
+                >
+                  <label id="multipleInputs"> colors example: red, blue </label>
```
Comment (jaxonL) [inline vue/configurator/src/components/Configurator.vue:103]: just caught this this time around -- if there are more config props with type = `ARRAY`, you would have multiple labels with the same id + placeholder test. that wouldn't be good for accessibility issues.

File: vue/configurator/src/components/Configurator.vue [hunk 4]
```diff
@@ -68,6 +68,72 @@
...
+            <el-row
+              v-else-if="prop.type === WidgetPropType.INTERFACE"
+              style="
+                align-items: center;
+                border-color: lightgrey;
+                border-style: solid;
+                border-width: thin;
+                border-radius: 5px;
+                height: 150px;
+                width: 250px;
```
Comment (jaxonL) [inline vue/configurator/src/components/Configurator.vue:81]: minor nit: can you add these styles to the scoped styles tag of this SFC ?

**Review by jaxonL — COMMENTED** [review]
minor nits remaining. overall, seems good to merge.

File: vue/configurator/src/components/Configurator.vue (hunk 4 above)
Comment (tingting-lwy) [reply vue/configurator/src/components/Configurator.vue]: Done! Added it in the scoped style by giving it a class called `prop-groupings` in case theres any future prop that has many variables to visually group together

File: vue/configurator/src/components/Configurator.vue (hunk 3 above)
Comment (tingting-lwy) [reply vue/configurator/src/components/Configurator.vue]: true! I actually decided I didn't need to ID this label since any labels created by config props with type = `ARRAY` would then be styled using the selector `label` rather than creating accessibility issues by potentially using same `ids`

File: vue/configurator/src/components/Configurator.vue [hunk 5]
```diff
@@ -321,4 +409,9 @@ export default class Configurator extends Vue {
   word-wrap: break-word;
   overflow: auto;
 }
+label {
```
Comment (jaxonL) [inline vue/configurator/src/components/Configurator.vue:412]: minor comment: if anyone were to add a `label` element to the configurator, it would be styled using the rules below.

**Review by jaxonL — APPROVED** [review]
